### ISSUE TITLE
Question about your code

### ISSUE THREAD

**Issue by tensorfreitas** [jhauswald/caffe-layer-extractor#1]
Hi thanks for your share in gitHub!!! 

I'm starting in Caffe and wanted to extract the features of a certain layer. 

in your code for every label:

`for(int d = 0; d < net->blobs()[i]->count(); ++d)
      ofs << net->blobs()[1]->cpu_data()[d] << "\n";`

shouldn't it be blobs()[i] instead of blobs[1] ?

Thank you for your time!

**Comment by jhauswald** [comment]
I wrote this a while ago but from what I remember Caffe only keeps a top and bottom blob to limit the amount of memory it consumes so there is a bottom blob (input to the layer) and and a top blob (output of the layer).

In the case of layers with weights, the top blob holds the weights. This is based on a certain version of Caffe (at the time I wrote this) so Caffe could have changed what it's been doing since.

Hope that helps!
